flaggers/duplicate.py

```diff
@@ -0,0 +1,29 @@
+from .flagger import Flagger, Flags, flaggers
+
+#Class implements duplicate check
+class Duplicate(Flagger):
```

Comment:
It looks like the duplicate flagger would have to be specially implemented, and can't be called from the same pipeline (since the normal flagger call would be looping through the flagger list and calling flagger.flag(row)). I don't think it's a problem, we predicted this would happen, but we should definitely document this.

Reply:
I made it so we can call if from the same pipeline. We just send row and full dataframe. This way we can append data flags normally (like we call each flagger for the row of data)

Reply:
I don't think it's a problem to pass in row and full dataframe, but you do have to change the interface and every flaggers to accomodate the full dataframe as well, as currently calling flaggers[i].flag(row, frame) would result in a too many arguments error.

Reply:
Also btw if you're going to go through with this, you should make sure that the argument names for the flag methods are consistent, i.e. all of them use `row` to mean data row and `data` to mean frames. Currently most flaggers use `data` to mean data row.

Reply:
True, but we can specifically check for the name of the flagger to accommodate for that, and in the case we are on duplicate flagger, send 2 args

Reply:
I guess, it's a bit ungraceful because if for example the sponsors wanted to add a new flagger that uses the data frame like the duplicate flagger does, they would also have to modify main to add another extra case. I think it's fine to change the interface to also include the data frame, but we can leave this as an exception if we don't think future flaggers won't need access to the data frame.

Reply:
I just can't think of any case (just out of the top of my head) in which sponsors would need to access the full dataframe again. Since it's more common to access 1 row at a time, I think it's better if we leave it as an exception.

Reply:
Sounds good.